amxmodx/newmenus.cpp

```diff
@@ -620,38 +620,45 @@ const char *Menu::GetTextString(int player, page_t page, int &keys)
 	LogError(amx, AMX_ERR_NATIVE, "Invalid menu id %d(%d)", p, g_NewMenus.length()); \
 	return 0; }
 
-//Makes a new menu handle (-1 for failure)
-//native csdm_makemenu(title[]);
+// native menu_create(const title[], const handler[], bool:ml = false);
 static cell AMX_NATIVE_CALL menu_create(AMX *amx, cell *params)
 {
-	int len;
-	char *title = get_amxstring(amx, params[1], 0, len);
-	validate_menu_text(title);
-	char *handler = get_amxstring(amx, params[2], 1, len);
+	enum args { arg_count, arg_title, arg_handler, arg_ml };
 
-	int func = registerSPForwardByName(amx, handler, FP_CELL, FP_CELL, FP_CELL, FP_DONE);
-	
-	if (func == -1)
+	int length;
+	const auto title    = get_amxstring(amx, params[arg_title], 0, length);
+	const auto handler  = get_amxstring(amx, params[arg_handler], 1, length);
+	const auto callback = registerSPForwardByName(amx, handler, FP_CELL, FP_CELL, FP_CELL, FP_DONE);
+
+	if (callback == -1)
 	{
-		LogError(amx, AMX_ERR_NOTFOUND, "Invalid function \"%s\"", handler);
+		LogError(amx, AMX_ERR_NOTFOUND, R"(Invalid function %s)", handler);
```

Comment:
Inner quotes are missing.

Reply:
It's defining a literal string with `R` to remove the need to escape.

Reply:
@Arkshine I mean `%s`.

Reply:
Oh, right.